src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -1383,7 +1383,7 @@ internal TdsOperationStatus TryReadInt32(out int value)
             {
                 // If the int isn't fully in the buffer, or if it isn't fully in the packet,
                 // then use ReadByteArray since the logic is there to take care of that.
-                TdsOperationStatus result = TryReadByteArray(buffer, 2);
+                TdsOperationStatus result = TryReadByteArray(buffer, 4);
```

Comment:
Sometimes I wonder how this has been working fine for so long ! 😄 
Thanks for catching and fixing this.

Reply:
This was a mistake by me when re-implementing the changes. It's correct in main. It caused some tests to fail so i could track it down and fix it.